File: capture-server/src/coach.cpp
#include "coach.h"
#include "gateway.h"

Coach::Coach() : EntityBase("coach") { }

void Coach::report()
{
    EntityBase::report();

    Gateway::instance().route("GET", "/api/coaches", // To request LIST
        [this](const Request &req, Response &rsp)
        {
            this->list(req, rsp);
        });

    Gateway::instance().route("GET", "/api/coach", // To request SINGLE
        [this](const Request &req, Response &rsp)
        {
            this->find(req, rsp);
        });

    Gateway::instance().route("POST", "/api/coach", // To request INSERT
        [this](const Request &req, Response &rsp)
        {
            this->create(req, rsp);
        });

    Gateway::instance().route("PUT", "/api/coach", // To request UPDATE
        [this](const Request &req, Response &rsp)
        {
            this->update(req, rsp);
        });

    Gateway::instance().route("DELETE", "/api/coach", // To request DELETE
        [this](const Request &req, Response &rsp)
        {
            this->remove(req, rsp);
        });
}

Coach::~Coach() { }

int Coach::id() const
{
    return this->m_model.get<int>("id");
}

std::string Coach::sport() const
{
    return this->m_model.get<std::string>("sport");
}

std::string Coach::title() const
{
    return this->m_model.get<std::string>("title");
}

std::string Coach::program() const
{
    return this->m_model.get<std::string>("program");
}

std::string Coach::year() const
{
    return this->m_model.get<std::string>("year");
}

std::string Coach::first_name() const
{
    return this->m_model.get<std::string>("first_name");
}

std::string Coach::last_name() const
{
    return this->m_model.get<std::string>("last_name");
}

std::string Coach::mobile() const
{
    return this->m_model.get<std::string>("mobile");
}

std::string Coach::email() const
{
    return this->m_model.get<std::string>("email");
}

File: capture-server-ui/src/app/services/models-interfaces/coach.ts
import { Data } from "./full-stack-interface";
import { DataBase } from "./model";

export class CoachData extends DataBase<Data.Coach> {
  
  
    public override get id() : number {
        return this._model.id;
   }
    public override set id(v : number) {
        this._model.id = v;
    }
    
    
    public get sport() : string {
        return this._model.sport;
    }
    public set sport(v : string) {
        this._model.sport = v;
    }
    
    
    public get title() : string {
        return this._model.title;
    }
    public set title(v : string) {
        this._model.title = v;
    }
    
    
    public get program() : string {
        return this._model.program;
    }
    public set program(v : string) {
        this._model.program = v;
    }
    
    
    public get year() : string {
        return this._model.year;
    }
    public set year(v : string) {
        this._model.year = v;
    }

    
    public get first_name() : string {
        return this._model.first_name;
    }
    public set first_name(v : string) {
        this._model.first_name = v;
    }
    
    public get last_name() : string {
        return this._model.last_name;
    }
    public set last_name(v : string) {
        this._model.last_name = v;
    }
    
    public get mobile() : string {
        return this._model.mobile;
    }
    public set mobile(v : string) {
        this._model.mobile = v;
    }
    
    public get email() : string {
        return this._model.email;
    }
    public set email(v : string) {
        this._model.email = v;
    }
    
    public get social_links() : Data.SocialLinks {
        return this._model.social_links;
    }
    public set social_links(v : Data.SocialLinks) {
        this._model.social_links = v;
    }
    
    
    public get picture() : string {
        return this._model.picture;
    }
    public set picture(v : string) {
        this._model.picture = v;
    }
    
    public get bio() : string {
        return this._model.bio;
    }
    public set bio(v : string) {
        this._model.bio = v;
    }
    
    
    public get stats() : string {
        return this._model.stats;
    }
    public set stats(v : string) {
        this._model.stats = v;
    }
    
}
      
